pkg/statefulsetmanager/sts_manager.go

```diff
@@ -77,7 +77,7 @@ func (sm *statefulSetManager) IsValidStatefulSetPod(ctx context.Context, namespa
 
 	stsName, replicas, found := getStatefulSetNameAndOrdinal(podName)
 	if !found {
-		return false, fmt.Errorf("failed to parse the name and replica of its StatefulSet controller from the name of Pod '%s/%s'", namespace, podName)
```

Comment:
Only StatefulSet kind corresponding pod will call this function. if we meet error after `getStatefulSetNameAndOrdinal `, which means that we meet a third-party controller StatefulSet. For the Spiderpool desgin, we regard all of third-party controller are unstateful